modules/users/client/components/ProfileViewBasics.component.js

```diff
@@ -13,9 +34,87 @@ export default function ProfileViewBasics({ profile }) {
         {(profile.seen) ? t('Online {{date, fromNow}}', { date: new Date(profile.seen) }) : t('Online long ago')}
       </span>
     </div>
+    {profile.locationLiving &&
+      <div className="profile-sidebar-section">
+        <i className="icon-fw icon-building text-muted"></i>
+        {t('Lives in')} <a ui-sref="search.map({location: profileCtrl.profile.locationLiving})"
+        >{profile.locationLiving}</a>
+      </div>}
+    {profile.locationFrom &&
+      <div className="profile-sidebar-section">
+        <i className="icon-fw icon-home text-muted"></i>
+        {t('From')} <a ui-sref="search.map({location: profileCtrl.profile.locationFrom})">
+          {profile.locationFrom}</a>
+      </div>}
+    {profile.languages.length > 0 &&
+      <div className="profile-sidebar-section">
+        <h4 id="profile-languages">{t('Languages')}</h4>
+        <ul className="list-unstyled" aria-describedby="profile-languages">
+          {profile.languages.map(code => <li key={code}>{t(languages[code], { ns: 'languages' }) || code}</li>)}
+        </ul>
+      </div>}
+      { (hasConnectedAdditionalSocialAccounts() || profile.extSitesBW || profile.extSitesCS || profile.extSitesWS) &&
+        <div className="profile-sidebar-section">
+          <h4 id="profile-networks" aria-label="Member in other networks">
+            {t('Elsewhere')}
+          </h4>
+          <ul className="social-profiles list-unstyled"
+            aria-describedby="profile-networks">
+            {/*
+              Facebook profile link is hidden here until issue with their API gets resolved
+              See https://github.com/Trustroots/trustroots/issues/237
+            */}
+            {profile.additionalProvidersData && Object.keys(profile.additionalProvidersData).map(
+              (key, indeks) =>
+              {return (key !== 'facebook') && <li className="social-profile" key={indeks}><i className={`social-profile-icon icon-fw icon-lg icon-${key}`}></i>
```

Comment:
Since this is a list with some unique values, we can use those unique values as list's key.

React supports using an index as a key, but it's recommended to avoid it if possible:

> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.

via https://reactjs.org/docs/lists-and-keys.html

In this specific case it probably wouldn't cause re-renderings in any situation but what do you think about doing something like this:

```jsx
Object.keys(profile.additionalProvidersData).map( (service) => {
  return service !== 'facebook' && (
    <li className="social-profile" key={network}>
      <i className={`social-profile-icon icon-fw icon-lg icon-${network}`}></i>
      ...
```

Reply:
Thanks for the article, it is interesting to know.
I changed it.